index.html

```diff
@@ -609,106 +609,211 @@
-    .catch((error) => { console.log("Adding watch failed: " + error.name )});
-
-    function processMessage(message) {
-      message.data.forEach((record) => {
-        if (record.recordType == "string") {
-          console.log('Data is string: ' + record.data);
-        } else if (record.recordType == 'json') {
-          processJSON(record.data);
-        } else if (record.recordType == 'url') {
-        console.log('Data is URL: ' + record.data);
-        } else if (record.recordType == 'opaque') {
-          processBinary(record.data);
-      });
-    };
 
-    function processBinary(data) {
-      console.log('Known (string) binary data: ');
-      console.log(String.fromCharCode.apply(null, new Uint16Array(data)));
-    };
+  <aside title="Push a text string to either a tag or peer"
+       class="example">
+    <p>
+      Pushing a text string (like the serialized JSON below)
+      to any kind of device is straight forward. Options can
+      be left out, as they default to pushing to both tags and
+      peers.
+    </p>
+    <pre class="highlight">
+navigator.nfc.push(
+    '{ prop1: "value1"; prop2: "value2" }'
+  ).then(() => {
+    console.log("Message pushed.");
+  }).catch((error) => {
+    console.log("Push failed :-( try again.");
+  });
+    </pre>
+  </aside>
 
-    function processJSON(data) {
-      var obj = JSON.parse(data);
-      console.log('JSON data: ' + obj.myProperty.toString());
-    };
-  </pre>
-  <pre title="Save and restore game progress with another device"
-       class="example highlight">
-    // Watching for specific content; here only own game data.
-    navigator.nfc.watch(reader, { url: "*/mypath/mygame/*" });
+  <aside title="Push a text string to a peer device" class="example">
+    <p>
+      It is possible to restrict to which devices (tags or peers) data
+      should be pushed. Below we specify to only push to peers,
+      and thus, no data is pushed when the user taps a tag.
+    </p>
+    <pre class="highlight">
+navigator.nfc.push(
+    "Text meant for peers only", { target: "peer" }
+  ).then(() => {
+    console.log("Message pushed.");
+  }).catch((error) => {
+    console.log("Push failed :-( try again.");
+  });
+    </pre>
+  </aside>
 
-    function reader(message) {
-      console.log("Game state received from: " + message.url);
-      console.log("Stored state: " + message.data);
+  <aside title="Push a URL to either a tag or peer" class="example">
+    <p>
+      In order to push an NDEF record of URL type, simply use NFCMessage.
+    </p>
+    <pre class="highlight">
+navigator.nfc.push({
+    data: [{ recordType: "url", data: "/path/resource/" }];
+  }).then(() => {
+    console.log("Message pushed.");
+  }).catch((error) => {
+    console.log("Push failed :-( try again.");
+  });
+    </pre>
+  </aside>
 
-      // Now do some calculations and then update the state.
+  <aside title="Read and update data stored on tag" class="example">
```

Comment:
This Example 4 still has the inline comments. Also Example 5. They should be cleansed, too.